BMS_Software_Main/BMS_Software_Main.ino

```diff
@@ -0,0 +1,170 @@
+#include "BMS_Software_Main.h"
+#include "RoveComm.h"
+//rovecomm and packet instances
+RoveCommEthernet RoveComm;
+rovecomm_packet packet; 
+
+//timekeeping variables
+uint32_t last_update_time;
+
+//declare the Ethernet Server in the top level sketch with the requisite port ID any time you want to use RoveComm
+EthernetServer TCPServer(RC_ROVECOMM_BMSBOARD_PORT);
+void setup() 
+{
+  Serial.begin(9600);   //
+  Serial3.begin(9600); //comms with LCD display
+  
+  RoveComm.begin(RC_BMSBOARD_FOURTHOCTET, &TCPServer);
+  delay(ROVECOMM_DELAY);
+  
+  setInputPins();
+  setOutputPins();
+  setOutputStates();
+}
+
+
+void loop() 
+{
+  //listen for rovecomm packets
+  packet = RoveComm.read();
+  
+  //if estop, go directly to estop function
+  if (packet.data_id = RC_BMSBOARD_BMSSTOP_DATA_ID)
+  {
+    estop(packet.data[0]);
+  }
+
+  //get telemetry and send it
+  float temp = getTemperature();
+  float cells[RC_BMSBOARD_CELLV_MEAS_DATA_COUNT] = getCellVoltages();
+  float pack_voltage = getPackVoltage(cells[RC_BMSBOARD_CELLV_MEAS_DATA_COUNT]);
+  float pack_current = getCurrent();
+  RoveComm.write(RC_BMSBOARD_PACKI_MEAS_DATA_ID, pack_current);
+  RoveComm.write(RC_BMSBOARD_PACKV_MEAS_DATA_ID, pack_voltage);
+  RoveComm.write(RC_BMSBOARD_CELLV_MEAS_DATA_ID, cells[]);
+  RoveComm.write(RC_BMSBOARD_TEMP_MEAS_DATA_ID, temp);
```

Comment:
We will link some places to look at correcting how this works and standardizing it with other places in C++ rovecomm - You can't return an array from a function, it needs to be pass by reference or a pointer (I would advise passing by reference, but we want to standardize with whatever we do most often). Similarly, this is not how you pass arrays into rovecomm.write(), and that needs to be corrected as well.

Reply:
Make sure whatever changes you do here to fix passing arrays around cascade down into the functions themselves.